internal/db/copy.go

```diff
@@ -42,3 +40,71 @@ func CopyAllObjects(ctx context.Context, tx pgx.Tx, source int64, target int64)
 
 	return nil
 }
+
+func CloneToProject(ctx context.Context, tx pgx.Tx, source int64, target int64, vrange VersionRange) (*int64, error) {
+	objectQuery := &pb.ObjectQuery{
+		Path:        "",
+		IsPrefix:    true,
+		WithContent: false,
+		WithHash:    true,
+	}
+
+	latestVersion, err := LockLatestVersion(ctx, tx, target)
+
+	if err != nil {
```

Comment:
Nit: The pattern in this codebase is to keep the error checks right under the line where the error is defined, instead of including an extra blank line.